iac-extensions/docker/src/main/java/org/sonar/iac/docker/checks/WeakHashAlgorithmsCheck.java

```diff
@@ -0,0 +1,156 @@
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.iac.docker.checks;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+import java.util.stream.Stream;
+import javax.annotation.Nullable;
+import org.sonar.check.Rule;
+import org.sonar.iac.common.api.checks.CheckContext;
+import org.sonar.iac.common.api.checks.IacCheck;
+import org.sonar.iac.common.api.checks.InitContext;
+import org.sonar.iac.common.api.tree.HasTextRange;
+import org.sonar.iac.common.api.tree.impl.TextRange;
+import org.sonar.iac.common.api.tree.impl.TextRanges;
+import org.sonar.iac.docker.symbols.ArgumentResolution;
+import org.sonar.iac.docker.tree.api.Argument;
+import org.sonar.iac.docker.tree.api.RunInstruction;
+
+@Rule(key = "S4790")
+public class WeakHashAlgorithmsCheck implements IacCheck {
+
+  private static final String MESSAGE = "Using weak hashing algorithms is security-sensitive.";
+  private static final Set<String> OPENSSL_SENSITIVE_SUBCOMMAND = Set.of("md5", "sha1", "rmd160", "ripemd160");
+  private static final Set<String> OPENSSL_SENSITIVE_DGST_OPTION  = Set.of("md2", "md4", "md5", "sha1", "ripemd160", "ripemd", "rmd160");
+
+  @Override
+  public void initialize(InitContext init) {
+    init.register(RunInstruction.class, WeakHashAlgorithmsCheck::checkRun);
+  }
+
+  private static void checkRun(CheckContext ctx, RunInstruction runInstruction) {
+    Command.parse(runInstruction.arguments()).stream()
+      .filter(WeakHashAlgorithmsCheck::isOpenSslCallSensitive)
+      .forEach(cmd -> ctx.reportIssue(cmd.textRange, MESSAGE));
+  }
+
+  private static boolean isOpenSslCallSensitive(Command command) {
+    return "openssl".equals(command.executable) && !command.parameters.isEmpty()
+      && (isOpenSslSensitiveSubCommand(command) || isOpenSslSensitiveDgstSubCommand(command));
+  }
+
+  private static boolean isOpenSslSensitiveSubCommand(Command command) {
+    return OPENSSL_SENSITIVE_SUBCOMMAND.contains(command.parameters.get(0));
+  }
+
+  private static boolean isOpenSslSensitiveDgstSubCommand(Command command) {
+    return "dgst".equals(command.parameters.get(0))
+      && OPENSSL_SENSITIVE_DGST_OPTION.stream().anyMatch(command.options::containsKey);
+  }
+
+  private static class Command {
```

Comment:
Introducing the `Command` class, which goal is to parse provided list of `Argument` to split it (`&&` and `|`) and transform it into a list of `Command` with executable/options/parameters + `textRange` to process them individually.

Reply:
Yeah, good idea. You also need to include `;` which is a valid delimiter. However, this "simple" split approach can be dangerous when we have nested elements. 
```
CMD foo(bar1 && bar2)
```
Here the split is incorrect.

Reply:
Mmhh indeed, we need at least for now to document such failing case.
This approach is too naïve to handle all cases properly, I'm thinking about it like a babystep to provide a solution with a good balance between time spent and detection/precision.
I will do more search/test to see what cases it can handle and if we can improve it simply.